eoc_journal/eoc_journal.py

```diff
@@ -78,13 +81,23 @@ class EOCJournalXBlock(StudioEditableXBlockMixin, XBlock):
         'selected_pb_answer_blocks',
     )
 
+    def _get_course_id(self):
+        """ Get a course ID if available """
+        return getattr(self.runtime, 'course_id', 'all')
+
     def student_view(self, context=None):
         """
         View shown to students.
         """
         context = context.copy() if context else {}
 
         context['display_name'] = self.display_name
+        user = self._get_current_user()
+        course_id = getattr(self.runtime, 'course_id', 'course_id')
+        course_id = unicode(_normalize_id(course_id))
+        anonymous_user_id = anonymous_id_for_user(user, course_id, save=False)
```

Comment:
Crap, I think this got in the way after I did the rebase on top of master. did not notice it.